app/helpers/application_helper/toolbar/middleware_domain_center.rb

```diff
@@ -16,4 +16,22 @@ class ApplicationHelper::Toolbar::MiddlewareDomainCenter < ApplicationHelper::To
                    ]
                  ),
                ])
+  button_group('middleware_server_operations', [
+                 select(
+                   :middleware_server_power_choice,
+                   'fa fa-power-off fa-lg',
+                   t = N_('Power'),
+                   t,
+                   :items => [
+                     button(
+                       :middleware_server_stop,
```

Comment:
Yes, that would be fine. I just did something similar for JDR. Here is how you add the privilege: https://github.com/israel-hdez/manageiq/commit/2e73a64327aec470b169f23d6071fbd96f3288d8. You then need to do `rake db:seed` to finish registering the new privilege in the database (else, the button will disappear because of the user not having the privilege).